internal/controller/openshiftbuild_controller.go

```diff
@@ -79,39 +90,51 @@ func (r *OpenShiftBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			Reason:  "Initializing",
 			Message: "Initializing Openshift Builds Operator",
 		})
-		if err := r.Client.Status().Update(ctx, openShiftBuild); err != nil {
+
+		patch := client.MergeFrom(openShiftBuild.DeepCopy()) // Get base for patch
+		if err := r.Client.Status().Patch(ctx, openShiftBuild, patch); err != nil {
 			logger.Error(err, "Failed to initialize status")
 			return ctrl.Result{}, err
 		}
 	}
 
-	// TODO: Add any specific cleanup logic
+	// Handle Deletion
 	if !openShiftBuild.DeletionTimestamp.IsZero() {
 		return ctrl.Result{}, r.HandleDeletion(ctx, openShiftBuild)
 	}
 
 	// Reconcile Shipwright Build
-	if err := r.ReconcileShipwrightBuild(ctx, openShiftBuild); err != nil {
-		logger.Error(err, "Failed to reconcile ShipwrightBuild")
+	if componentErr := r.ReconcileShipwrightBuild(ctx, openShiftBuild); componentErr != nil {
```

Comment:
maybe a nitpick but you can name the error shipwrightErr and sharedResourcesErr for clarity